base/applications/cmdutils/more/more.c

```diff
@@ -134,28 +356,113 @@ PagePrompt(PCON_PAGER Pager, DWORD Done, DWORD Total)
      */
     ConClearLine(Pager->Screen->Stream);
 
+    switch (s_chSubCommand)
+    {
+        case L'P':
+            s_chSubCommand = 0;
+            Pager->ScrollRows = nLines;
+            return TRUE;
+        case L'S':
+            s_chSubCommand = 0;
+            s_dwFlags |= FLAG_PLUSn;
+            s_nNextLineNo = Pager->lineno + nLines;
+            Pager->ScrollRows = Pager->ScreenRows - 1;
+            return TRUE;
+        case L'-':
+            s_chSubCommand = 0;
+            Pager->PagerAction = MorePageActionDoNothing;
+            return TRUE;
+    }
+    s_chSubCommand = 0;
+
     /* Ctrl+C or Ctrl+Esc: Control Break */
-    if ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
-        ((KeyEvent.wVirtualKeyCode == L'C') &&
-         (KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED))))
+    if (fCtrl && ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
+                  (KeyEvent.wVirtualKeyCode == L'C')))
     {
         /* We break, output a newline */
         WCHAR ch = L'\n';
         ConStreamWrite(Pager->Screen->Stream, &ch, 1);
         return FALSE;
     }
 
-    /* 'Q': Quit */
-    // FIXME: Available only when command extensions are enabled.
-    if ((KeyEvent.wVirtualKeyCode == L'Q') &&
-        !(KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED)))
+    if (fCtrl)
     {
-        /* We break, output a newline */
-        WCHAR ch = L'\n';
-        ConStreamWrite(Pager->Screen->Stream, &ch, 1);
-        return FALSE;
+        /* Do nothing */
+        Pager->PagerAction = MorePageActionDoNothing;
+        return TRUE;
     }
 
+    /* [Space] key: One page */
+    if (KeyEvent.wVirtualKeyCode == VK_SPACE)
+    {
+        if (s_dwFlags & FLAG_C)
+        {
+            /* Clear the screen */
+            ConClearScreen(Pager->Screen);
+        }
+        Pager->ScrollRows = Pager->ScreenRows - 1;
+        return TRUE;
+    }
+
+    /* If extended features are available */
+    if (s_dwFlags & FLAG_E)
+    {
+        /* 'Q': Quit */
+        if (KeyEvent.wVirtualKeyCode == L'Q')
+        {
+            /* We break, output a newline */
+            WCHAR ch = L'\n';
+            ConStreamWrite(Pager->Screen->Stream, &ch, 1);
+            return FALSE;
+        }
+
+        /* 'F': Next file */
+        if (KeyEvent.wVirtualKeyCode == L'F')
+        {
+            Pager->PagerAction = MorePageActionNextFile;
+            return TRUE;
+        }
+
+        /* '?': Show Options */
+        if (KeyEvent.uChar.UnicodeChar == L'?')
+        {
+            Pager->PagerAction = MorePageActionDoNothing;
+            s_nPromptID = IDS_CONTINUE_OPTIONS;
+            return TRUE;
```

Comment:
Does the following work? :
```suggestion
            s_nPromptID = IDS_CONTINUE_OPTIONS;
            goto top;
```

Reply:
OK.